be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -84,6 +84,9 @@ OLAPStatus BetaRowsetWriter::_add_row(const RowType& row) {
     if (PREDICT_FALSE(_segment_writer == nullptr)) {
         RETURN_NOT_OK(_create_segment_writer());
     }
+    if (PREDICT_FALSE(_segment_writer->reach_capacity())) {
```

Comment:
You can check with _max_segment_size together.

Reply:
+1